src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -701,6 +706,53 @@ impl WorkunitStore {
         });
     }
   }
+
+  ///
+  /// Records an observation of a time-like metric into a histogram.
+  ///
+  pub fn record_observation(&self, metric: ObservationMetric, value: u64) {
+    let mut histograms_by_metric = self.observation_data.observations.lock();
+    histograms_by_metric
+      .entry(metric)
+      .and_modify(|h| {
+        let _ = h.record(value);
+      })
+      .or_insert_with(|| {
+        let mut h = hdrhistogram::Histogram::<u64>::new(3).expect("Failed to allocate histogram");
```

Comment:
This defaults histograms to three significant digits which is what the library recommends as a default. I believe the range is 1-5.

It also defaults to an auto-resizing data structure. There are constructors that can take (min, max) range which allocate upfront and do not reallocate later. I think it should be fine this way for now until we have metrics that we would know the range for.